src/rpc/blockchain.cpp

```diff
@@ -1773,8 +1772,8 @@ static RPCHelpMan getblockstats()
         }
     }
 
-    const CBlock block = GetBlockChecked(chainman.m_blockman, pindex);
-    const CBlockUndo blockUndo = GetUndoChecked(chainman.m_blockman, pindex);
+    const CBlock block = GetBlockChecked(chainman.m_blockman, &pindex);
+    const CBlockUndo blockUndo = GetUndoChecked(chainman.m_blockman, &pindex);
```

Comment:
suggestion, these are memoizing not-cheap-to-copy class objects, so may as well avoid doing a copy 
```diff
-    const CBlock block = GetBlockChecked(chainman.m_blockman, &pindex);
-    const CBlockUndo blockUndo = GetUndoChecked(chainman.m_blockman, &pindex);
+    const CBlock& block = GetBlockChecked(chainman.m_blockman, &pindex);
+    const CBlockUndo& blockUndo = GetUndoChecked(chainman.m_blockman, &pindex);
```